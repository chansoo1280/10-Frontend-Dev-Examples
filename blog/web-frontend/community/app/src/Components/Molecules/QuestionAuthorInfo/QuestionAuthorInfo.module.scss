@import "@Styles/scss/default.scss";

.question-author-info {

    &__date {
        padding-left: 16px;
        color: $black-25;
    }
}