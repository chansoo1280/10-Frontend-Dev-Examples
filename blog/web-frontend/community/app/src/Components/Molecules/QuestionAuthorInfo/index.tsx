// #region Global Imports
import React from "react"
import classNames from "classnames"
import Image, { ImageLoader } from "next/image"
// #endregion Global Imports

// #region Local Imports
import { defaultProps, Button, Text } from "@Components"
import styles from "./QuestionAuthorInfo.module.scss"
import { dateFormat } from "@Utils"
// #endregion Local Imports
const myLoader: ImageLoader = ({ src, width, quality }) => {
    return `/public/Images/${src}?w=${width}&q=${quality || 75}`
}
interface QuestionAuthorInfoProps extends defaultProps {
    userName: string
    created?: string
}

const QuestionAuthorInfo = (props: QuestionAuthorInfoProps): JSX.Element => {
    const { userName, created, show, className, ...rest } = props
    const prefixCls = "question-author-info"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )
    return (
        <Text className={classes} size="small" {...rest}>
            <Button size="small" type="link">
                <Image loader={myLoader} src="me.png" alt="Picture of the user" width={24} height={24} />
                {userName}
            </Button>
            published at{" "}
            <Text type="link" size="small">
                https://ant.design
            </Text>
            <Text show={created !== undefined} size="small" className={classNames(styles[`${prefixCls}__date`])}>
                {/* 2021-02-05 13:51 */}
                {dateFormat(new Date(created || ""), "yyyy-MM-dd hh:mm")}
            </Text>
        </Text>
    )
}
export default QuestionAuthorInfo
