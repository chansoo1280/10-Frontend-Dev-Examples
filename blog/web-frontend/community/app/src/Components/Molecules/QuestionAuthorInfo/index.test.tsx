import { fireEvent, render, screen } from "@testing-library/react"
import QuestionAuthorInfo from "."
describe("QuestionAuthorInfo", () => {
    it("renders questionAuthorInfo unchanged", () => {
        const { container } = render(<QuestionAuthorInfo userName={""} created={""} />)
        expect(container).toMatchSnapshot()
    })
    it("should render a questionAuthorInfo with the class", () => {
        render(<QuestionAuthorInfo data-testid="questionAuthorInfo" userName={""} created={""} />)

        const questionAuthorInfo = screen.getByTestId("questionAuthorInfo")
        expect(questionAuthorInfo).toBeInTheDocument()
        expect(questionAuthorInfo).toHaveClass("text", "text--small", "question-author-info")
    })
})
