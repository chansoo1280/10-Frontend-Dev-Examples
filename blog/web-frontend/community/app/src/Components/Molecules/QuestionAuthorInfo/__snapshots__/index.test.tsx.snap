// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionAuthorInfo renders questionAuthorInfo unchanged 1`] = `
<div>
  <span
    class="text text--small question-author-info"
  >
    <button
      class="btn btn--small btn--link"
      data-click-animating="false"
      type="button"
    >
      <img
        alt="Picture of the user"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/public/Images/me.png?w=48&q=75"
        srcset="/public/Images/me.png?w=32&q=75 1x, /public/Images/me.png?w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
    </button>
    published at
     
    <span
      class="text text--small text--link"
    >
      https://ant.design
    </span>
    <span
      class="text text--small question-author-info__date"
    />
  </span>
</div>
`;
